### DISCUSSION TITLE
How to make i3 more laptop friendly

### DISCUSSION THREAD

**Discussion by 70dakhaled** [i3/i3#5831] [Q&A]
I need help of a beginner guide to making i3 more usable on laptop
I'm new to i3 i learned the keybindings ,and it's amazing when used on desktop ,but for laptop i feele there is some inconveniencies

most of what i need from bar is:  l basic features like ,language ,wifi network , brightness control  ,bluetooth  , and i would be thankful if someone help lead me to a guide or tutorial to read .

my problems:
1- i couldn't use brightness keys on laptop keyboard i tried making keybidndings didn't work but im sure i've not done it right 
2- network mangement  when i used i3 on top of ubuntu the network applet was working well but when i tried using i3  only then tried to install network management and applet didn't work .

**Comment by vishal340** [comment]
install brightnessctl. then use it in volume_brightness.sh file.

**Reply by 70dakhaled** [reply]
noted  thank you ,will try it since brightnessctl worked in terminal

**Comment by okraits** [comment]
1. I can recommend i3blocks instead of i3status. i3status is nice but i3blocks is more flexible and the configuration is a bit easier.
2. wifi management: did you start nm-applet? What was the exact problem?
3. brightness control: you might need a specific tool for your laptop. The Arch Linux wiki has a good overview: https://wiki.archlinux.org/title/Backlight#Backlight_utilities. Make sure that it works from a terminal, then you can map it to your keys in your config.

**Reply by 70dakhaled** [reply]
I did try it in terminal  and it worked using using brightnessctl thank you for the tips  <3
i did start it maybe there was some package missing  it seemed like as if there was a network manager that is working i didn't know  how and why the applet not reading it correctly .
i installed it on bare debain install without  some stuff preconfigured and im ignorant  about   how to set up i3 from beginning and required packages that are setup by default in desktop environments. 
i'm using garuda i3 jsut to kickstart  and use it ,  keybindings are bit different i kinda liked default and the tabs  feature is so awesome  so will definetly get back to it .

**Comment by mikeredev** [comment]
i also recommend i3blocks, you can find my own implementation [here](https://github.com/mikeredev/i3blocks)

to manage brightness via i3 there is a script [here](https://github.com/mikeredev/dotfiles/blob/main/scripts/tools/manage-brightness.py)

you can run `xev` and then hit the relevant key (e.g., `XF86MonBrightnessUp`) to get a string you can copy into your i3 config's keybinding

i dont use nm-applet but if you cant get it working it's not hard to build an i3 blocklet that shows the same

**Reply by 70dakhaled** [reply]
thank you so much fo your help  your configs look awesome

**Reply by 70dakhaled** [reply]
yes especially i don't need much at the beginning just need to get used to it  and learn the configs. 
 maybe use aspiration from others as a guide or refrence  to choose what might suit me .

**Reply by mikeredev** [reply]
you're welcome, i think starting from a working config someone else provides is ok you'll just be missing out on the benefits of i3 if you dont learn how to configure your own. once you're comfortable with the environment and how the bar is used take some time to read the [user guide](https://i3wm.org/docs/userguide.html) and see what functionality it has you might be interested in using
